pipeline.py

```diff
@@ -72,31 +88,82 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+        for article in articles:
+            raw_text = article.get_raw_text()
+            tokens = self._process(raw_text)
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            for token in tokens:
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+            article.save_as(*cleaned_tokens, kind=ArtifactType.cleaned)
+            article.save_as(*single_tagged_tokens, kind=ArtifactType.single_tagged)
+            print(*cleaned_tokens)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        mystem = Mystem()
+        analyzed_text = mystem.analyze(raw_text)
+        tokens = []
+        for analyzed_word in analyzed_text:
+            if not analyzed_word.get('analysis'):
+                continue
+            morphological_token = MorphologicalToken(original_word=analyzed_word.get('text'))
```

Comment:
all pieces are very fragile - you might have empty dictionaries and what happens then?

Reply:
or analysis lists might be empty